### PR TITLE
feat: ask AI feature

### PR DESCRIPTION
**Pull request by joaopcm**

### CODE REVIEW

File: src/components/WebContainerEditor.tsx [hunk 1]
```diff
@@ -22,13 +25,20 @@ const CodeEditor = dynamic(
...
   const { codeSnippet, setCodeSnippet } = useEditor()
+  const router = useRouter()
+
+  const id = router.query.id as string
+  const isSavedNotePage = typeof id === 'string'
 
   useEffect(() => {
     return () => {
       setOutput([])
+      setAIExplanation(undefined)
```
Comment (NicolasLopes7) [inline src/components/WebContainerEditor.tsx:39]: What happens if we don't cleanup?

**Comment by joaopcm** [comment]
@NicolasLopes7, do you think we should provide the OpenAI API Key?

I ask this because OpenAI gives $18 for testing. I am currently using this budget to make things work in this PR. To get around this, I thought of the following flow:
1. User clicks on the "Ask AI" button;
2. We check if the answer is not already saved in Redis;
3. If not, we open a modal for the user to provide his token;
4. We save the user's API Key in local storage;
5. We use the provided API key to consume the AI;

Which option do you think is better?

File: src/components/WebContainerEditor.tsx (hunk 1 above)
Comment (joaopcm) [reply src/components/WebContainerEditor.tsx:39]: When changing the page, the `WebContainerEditor` renders the output and the AI explanation automatically (from the last visited page), even though we don't press any button in the current route.

File: src/pages/api/ask-ai.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      return res.status(400).json({
+        message: 'Missing code snippet',
+      })
+    }
+
+    if (!req.body.id) {
+      return res.status(400).json({
+        message: 'Missing ID',
+      })
+    }
```
Comment (NicolasLopes7) [inline src/pages/api/ask-ai.ts:26]: might add zod to the `Tech Debt` section on README

Comment (joaopcm) [reply src/pages/api/ask-ai.ts:26]: Also, it'd be nice to have a error handler in the client side. Will do it. Tks!

**Review by NicolasLopes7 — APPROVED** [review]
Looks good overall! I think we can create another PR refactoring the whole AI thing to a service/hook/anything. I think [zustand's store](https://zustand-demo.pmnd.rs/) would be agreat approach.

File: src/components/WebContainerEditor.tsx [hunk 3]
```diff
@@ -22,13 +25,20 @@ const CodeEditor = dynamic(
 export function WebContainerEditor() {
   const [output, setOutput] = useState<string[]>([])
   const [isRunning, setIsRunning] = useState(false)
+  const [aiExplanation, setAIExplanation] = useState<string>()
+  const [askingAI, setAskinAI] = useState(false)
   const { codeSnippet, setCodeSnippet } = useEditor()
+  const router = useRouter()
+
+  const id = router.query.id as string
+  const isSavedNotePage = typeof id === 'string'
```
[the remark this answers is not in the document]
Comment (joaopcm) [reply src/components/WebContainerEditor.tsx:34]: Added to tech debits list :)

File: src/pages/api/ask-ai.ts (hunk 2 above)
Comment (joaopcm) [reply src/pages/api/ask-ai.ts:26]: Added to tech debt list :)

**Comment by joaopcm** [comment]
> Looks good overall! I think we can create another PR refactoring the whole AI thing to a service/hook/anything. I think [zustand's store](https://zustand-demo.pmnd.rs/) would be agreat approach.

@NicolasLopes7, I just started a discussion to talk about `zustand's store` on Nodepad [here](https://github.com/joaopcm/nodepad/discussions/19) :)

star-wars-dancing-gif

**Pull request merged**